### PR TITLE
M2-6127, M2-6128: Unit tests for ConstructCompletionsService

### PR DESCRIPTION
**Pull request by anq83**
### 📝 Description

🔗 [Jira Ticket M2-6127](https://mindlogger.atlassian.net/browse/M2-6127)
🔗 [Jira Ticket M2-6128](https://mindlogger.atlassian.net/browse/M2-6128)

Changes include:

- Unit tests for ConstructCompletionsService.constructForIntermediate
- Unit tests for ConstructCompletionsService.constructForFinish
- Unit tests for associated with ConstructCompletionsService external operations

### CODE REVIEW

File: src/widgets/survey/model/tests/survey-operations.test.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
+import { AvailabilityType, PeriodicityType } from '@app/abstract/lib';
+import { ScheduleEvent } from '@app/entities/event';
+import { Answers } from '@app/features/pass-survey';
+import {
+  getDrawerItem,
+  getDrawerResponse,
+  getSliderItem,
+  getSplashItem,
+  getTextInputItem,
```
Comment (anq83) [inline src/widgets/survey/model/tests/survey-operations.test.ts:9]: add mock postfix?

File: src/widgets/survey/model/services/tests/testHelpers.ts [hunk 2]
```diff
@@ -128,3 +140,108 @@ export const deleteLogAvailableTo = (
...
+          startAt: 12367800000,
+          endAt: null,
+        },
+      },
+    },
+  };
+  return progress;
+};
+
+export const getActivityRecordMockResult = (
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/testHelpers.ts:231]: rename?

File: src/widgets/survey/model/services/tests/testHelpers.ts [hunk 3]
```diff
@@ -128,3 +140,108 @@ export const deleteLogAvailableTo = (
...
+          value: answersMock[0],
+        },
+      },
+    },
+    {
+      type: 'NEXT',
+      payload: {
+        activityId: 'mock-activity-id-1',
+        activityItemId: 'mock-slider-item-1',
+        date: addMinutes(new Date(2023, 3, 5), 1).getTime(),
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/testHelpers.ts:163]: use mocked now?

File: src/widgets/survey/model/tests/survey-operations.test.ts [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+  }),
+);
+
+const writeFileMock = jest.fn().mockResolvedValue([undefined]);
+
+jest.mock('@app/entities/drawer/lib/utils/SvgFileManager', () => ({
+  writeFile: (_: string, svg: string) => writeFileMock(svg),
+}));
+
+const getMockEvent = (): ScheduleEvent => {
```
Comment (anq83) [inline src/widgets/survey/model/tests/survey-operations.test.ts:39]: should be somewhere in testHelpers

File: src/widgets/survey/model/tests/survey-operations.test.ts [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+    const result = getUserIdentifier(items, answers);
+
+    expect(result).toEqual(undefined);
+  });
+
+  it('"getItemIds" should return items identifiers excluding the ones that cannot have answer', () => {
+    const items = [
+      getSplashItem('item-id-1'),
+      getTextInputItem('item-id-2'),
```
Comment (anq83) [inline src/widgets/survey/model/tests/survey-operations.test.ts:145]: shouldn't it be in result?

File: src/widgets/survey/model/tests/survey-operations.test.ts [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    const answers: Answers = {
+      0: answer1,
+      1: answer2,
+      2: answer3,
+    };
+
+    const answerDtos = mapAnswersToDto(items, answers);
+
+    const result = fillNullsForHiddenItems(
+      ['item-id-1', 'item-id-2', 'item-id-3', 'item-id-4'],
```
Comment (anq83) [inline src/widgets/survey/model/tests/survey-operations.test.ts:177]: remove id-4

File: src/widgets/survey/model/tests/survey-operations.test.ts [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+      ],
+    );
+
+    expect(result).toEqual({
+      answers: [null, { value: '5' }, 'mock-identity'],
+      itemIds: ['item-id-1', 'item-id-2', 'item-id-3'],
+    });
+  });
+
+  it('"createSvgFiles" should call SvgFileManager.writeFile with scgStrings taken from answers', async () => {
```
Comment (anq83) [inline src/widgets/survey/model/tests/survey-operations.test.ts:204]: check

File: src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts [hunk 8]
```diff
@@ -0,0 +1,629 @@
...
+    };
+
+    const itemsMock: PipelineItem[] = [
+      getSliderItem('mock-slider-item-1', 'mock-slider-id-1'),
+      getSliderItem('mock-slider-item-2', 'mock-slider-id-2'),
+    ];
+
+    const userActionsMock: UserAction[] = getUserActionsMock(answersMock);
+
+    const getActivityRecordMock = jest
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts:135]: to function?

File: src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts [hunk 9]
```diff
@@ -0,0 +1,629 @@
...
+    const service = new ConstructCompletionsService(
+      saveSummaryMock,
+      {} as any,
+      progress,
+      pushToQueueMock,
+      dispatchMock,
+    );
+
+    //@ts-expect-error
+    service.queryDataUtils.getAppletDto = jest.fn().mockReturnValue({
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts:158]: to function

File: src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts [hunk 10]
```diff
@@ -0,0 +1,629 @@
...
+      encryption: 'applet-encryption-mock-1',
+      displayName: 'mock-applet-name-1',
+    });
+
+    await service.construct({
+      completionType: 'intermediate',
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts:164]: get input to function ?

File: src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts [hunk 11]
```diff
@@ -0,0 +1,629 @@
...
+    expect(getActivityRecordMock).toBeCalledTimes(1);
+
+    expect(createSvgFilesMock).toBeCalledTimes(1);
+
+    expect(saveSummaryMock).toBeCalledTimes(0);
+
+    expect(mapAnswersToDtoMock).toBeCalledTimes(1);
+
+    expect(mockFillNullsForHiddenItems).toBeCalledTimes(1);
+
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts:182]: remove lines

File: src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts [hunk 12]
```diff
@@ -0,0 +1,629 @@
...
+      executionGroupKey: 'mock-flow-group-key-1',
+      flowId: 'mock-flow-id-1',
+      isFlowCompleted: false,
+      itemIds: ['mock-slider-id-1', 'mock-slider-id-2'],
+      logActivityName: 'mock-activity-name-1',
+      logCompletedAt: mockNowDate.toUTCString(),
+      scheduledTime: 1245800000,
+      startTime: 12389100000,
+      tzOffset: 3,
+      userActions: [],
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts:223]: add converted to dtos here?

File: src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts [hunk 13]
```diff
@@ -0,0 +1,629 @@
...
+
+    const userActionsMock: UserAction[] = getUserActionsMock(answersMock);
+
+    const recordMock = getActivityRecordMockResult(
+      answersMock,
+      userActionsMock,
+      itemsMock,
+    )!;
+
+    jest.spyOn(storageHelpers, 'getActivityRecord').mockReturnValue({
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts:248]: consider move to function

File: src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts [hunk 14]
```diff
@@ -0,0 +1,629 @@
...
+
+    const itemsMock: PipelineItem[] = [
+      getSliderItem('mock-slider-item-1', 'mock-slider-id-1'),
+      getSliderItem('mock-slider-item-2', 'mock-slider-id-2'),
+    ];
+
+    const userActionsMock: UserAction[] = getUserActionsMock(answersMock);
+
+    const getActivityRecordMock = jest
+      .spyOn(storageHelpers, 'getActivityRecord')
```
Comment (anq83) [inline src/widgets/survey/model/services/tests/ConstructCompletionsService.test.ts:322]: to function

**Review by BamMironov — APPROVED** [review]

**Pull request merged**